csharp/LogicalTypeFactory.cs

```diff
@@ -134,6 +134,11 @@ public virtual unsafe (Type physicalType, Type logicalType) GetSystemTypes(Colum
                 return (typeof(ByteArray), typeof(byte[]));
             }
 
+            if (logicalType.Type == LogicalTypeEnum.String)
```

Comment:
This is here because we don't return earlier in the function because the default mapping doesn't match the column descriptor's required value.